contracts/eden/src/eden-micro-chain.cpp

```diff
@@ -390,16 +407,68 @@ struct Status
 };
 EOSIO_REFLECT2(Status, active, community, initialMembers, genesisVideo, auctionDuration, memo)
 
+struct ElectionGroup;
+constexpr const char ElectionGroupConnection_name[] = "ElectionGroupConnection";
+constexpr const char ElectionGroupEdge_name[] = "ElectionGroupEdge";
+using ElectionGroupConnection = clchain::Connection<
+    clchain::ConnectionConfig<ElectionGroup, ElectionGroupConnection_name, ElectionGroupEdge_name>>;
+
+struct Election
+{
+   const election_object* obj;
+
+   eosio::block_timestamp time() const { return obj->time; }
+
+   ElectionGroupConnection groupsByRound(std::optional<uint8_t> gt,
+                                         std::optional<uint8_t> ge,
+                                         std::optional<uint8_t> lt,
+                                         std::optional<uint8_t> le,
+                                         std::optional<uint32_t> first,
+                                         std::optional<uint32_t> last,
+                                         std::optional<std::string> before,
+                                         std::optional<std::string> after) const;
+};
+EOSIO_REFLECT2(Election,
+               time,
+               method(groupsByRound, "gt", "ge", "lt", "le", "first", "last", "before", "after"))
+
 struct ElectionGroup
 {
    const election_group_object* obj;
 
    uint64_t id() const { return obj->id._id; }
+   Election election() const { return {&get<by_pk>(db.elections, obj->election_time)}; }
    uint8_t round() const { return obj->round; }
    auto winner() const { return get_member(obj->winner); }
    std::vector<Vote> votes() const;
 };
-EOSIO_REFLECT2(ElectionGroup, id, round, winner, votes)
+EOSIO_REFLECT2(ElectionGroup, id, election, round, winner, votes)
+
+ElectionGroupConnection Election::groupsByRound(std::optional<uint8_t> gt,
+                                                std::optional<uint8_t> ge,
+                                                std::optional<uint8_t> lt,
+                                                std::optional<uint8_t> le,
+                                                std::optional<uint32_t> first,
+                                                std::optional<uint32_t> last,
+                                                std::optional<std::string> before,
+                                                std::optional<std::string> after) const
+{
+   return clchain::make_connection<ElectionGroupConnection, ElectionGroupByRoundKey>(
+       gt ? std::optional{ElectionGroupByRoundKey{obj->time, *gt, ~uint64_t(0)}}           //
+          : std::nullopt,                                                                  //
+       ge ? std::optional{ElectionGroupByRoundKey{obj->time, *ge, 0}}                      //
+          : std::optional{ElectionGroupByRoundKey{obj->time, 0, 0}},                       //
```

Comment:
This relies on the fact that `gt` has higher priority that `ge`, which is not exactly obvious.